src/bsoncxx/test/bson_builder.cpp

```diff
@@ -1234,4 +1244,526 @@ TEST_CASE("builder::stream::document throws on consecutive keys", "[bsoncxx::bui
     REQUIRE_NOTHROW(doc << "far");
     REQUIRE_THROWS_AS(doc << "boo", bsoncxx::exception);
 }
+
+TEST_CASE("list builder appends utf8", "[bsoncxx::builder::list]") {
+    bson_t expected;
+    bson_init(&expected);
+
+    bson_append_utf8(&expected, "hello", -1, "world", -1);
+
+    SECTION("works with string literals") {
+        builder::list b{"hello", "world"};
+
+        bson_eq_object(&expected, b.view().get_document().value);
+    }
+
+    SECTION("works with std::string") {
+        builder::list b{"hello", std::string{"world"}};
+
+        bson_eq_object(&expected, b.view().get_document().value);
+    }
+
+    SECTION("works with b_utf8") {
+        builder::list b{"hello", types::b_utf8{"world"}};
+
+        bson_eq_object(&expected, b.view().get_document().value);
+    }
+
+    SECTION("works with const char*") {
+        const char* world = "world";
+        builder::list b{"hello", world};
+
+        bson_eq_object(&expected, b.view().get_document().value);
+    }
+
+    SECTION("works with char*") {
+        char* world = const_cast<char*>("world");
+        builder::list b{"hello", world};
+
+        bson_eq_object(&expected, b.view().get_document().value);
+    }
+
+    bson_destroy(&expected);
+}
+
+TEST_CASE("list builder appends double", "[bsoncxx::builder::list]") {
+    bson_t expected;
+    bson_init(&expected);
+
+    bson_append_double(&expected, "foo", -1, 1.1);
+
+    SECTION("works with raw float") {
+        builder::list b{"foo", 1.1};
+
+        bson_eq_object(&expected, b.view().get_document().value);
+    }
+
+    SECTION("works with b_double") {
+        builder::list b{"foo", types::b_double{1.1}};
+
+        bson_eq_object(&expected, b.view().get_document().value);
+    }
+
+    bson_destroy(&expected);
+}
+
+TEST_CASE("list builder appends binary", "[bsoncxx::builder::list]") {
+    bson_t expected;
+    bson_init(&expected);
+
+    bson_append_binary(&expected, "foo", -1, BSON_SUBTYPE_BINARY, (uint8_t*)"deadbeef", 8);
```

Comment:
"deadbeef" here is a little misleading. That is the raw bytes for the ASCII characters 'd', 'e', 'a', 'd', 'b', 'e', 'e', 'f', and is not interpreted as hex values. I suggest just making this the string "data" or something.